ragna/assistants/_anthropic.py

```diff
@@ -36,12 +36,50 @@ def _instructize_system_prompt(self, sources: list[Source]) -> str:
             + "</documents>"
         )
 
-    async def answer(
-        self, messages: list[Message], *, max_new_tokens: int = 256
+    def _render_prompt(self, prompt: Union[str, list[Message]]) -> Union[str, list]:
+        """
+        Ingests ragna messages-list or a single string prompt and converts to assistant-appropriate format.
+
+        Returns:
+            ordered list of dicts with 'content' and 'role' keys
+        """
+        if isinstance(prompt, str):
+            return [
+                {
+                    "content": prompt,
+                    "role": "user",
+                }
+            ]
```

Comment:
Instead of this pattern of handling strings and messages completely different, I would like to just wrap the prompt into a `Message` and proceed on the same branch from there on. Like

```python
if isinstance(prompt, str):
     messages = [Message(content=prompt, role=MessageRole.USER)]
else:
     messages = prompt

...
```

Reply:
Hrm - the notion here was for _render_prompt() to convert from `list[Message]`s into whatever form the particular assistant would actually process at the API call level - e.g. here the anthropic 'messages' won't be 1:1 aligned with ragna list[Message]

Called https://github.com/Quansight/ragna/blob/82637dda6450184ec411ce55d0442dc87ffb6496/ragna/assistants/_anthropic.py#L95 right before it gets sent in to the API end-point.

In your proposal, would we convert any prompt into list[Message] and then from that into final API call format? If so, then I'm in agreement and will make the change.

Reply:
Yes that is what I meant. I want to avoid having two branches with different return logic. The `prompt: str` input is just for convenience. We just put it into the "right" format and process everything after it the same way.

The output is of course dependent on the assistant and does not need to follow a common format. That is the reason why we are not putting this on the `Assistant` class as there is no standardization possible apart from our convention of calling the method `_render_prompt` everywhere.